addons/survey/tests/test_survey.py

```diff
@@ -222,6 +222,86 @@ def test_answer_validation_char_box_length(self):
             {}
         )
 
+    @users('survey_manager')
+    def test_simple_choice_validation_multiple_answers(self):
+        """
+        Check that a 'simple_choice' question fails validation if more than one
+        valid answer is provided.
+        """
+        question = self._add_question(
+            self.page_0, 'Simple Choice Constraint Test', 'simple_choice',
+            constr_mandatory=True,
+            comments_allowed=True,
+            comment_count_as_answer=True,
+            labels=[{'value': 'Choice X'}, {'value': 'Choice Y'}]
+        )
+        answer_choice_x_id = question.suggested_answer_ids[0].id
+        answer_choice_y_id = question.suggested_answer_ids[1].id
+
+        scenarios = [
+            (
+                'Two selected choices should not be allowed',
+                [answer_choice_x_id, answer_choice_y_id],
+                None,
+                True,
+            ),
+            (
+                'One choice and one comment that counts as an answer',
+                [answer_choice_x_id],
+                'This is my comment, which is also an answer.',
+                True,
+            ),
+            (
+                'A single valid answer should pass validation',
+                answer_choice_x_id,
+                None,
+                False,
+            ),
+            (
+                'A single valid comment should pass validation',
+                '',
+                'This is my comment, which is also an answer.',
+                False,
+            ),
+        ]
+
+        for case_description, answer, comment, is_multiple_answers in scenarios:
+            with self.subTest(case_description):
```

Comment:
Almost 😄 

Referring to earlier comment:

```py
with self.subTest(answers=answers, comment=comment):
    ....
```

> As for the variables of the subTest, it's usually the varying parameters of your assertion, which here matches the answers and the comment.

(Side-note: would also name second variable "answerS" plural as it can contain multiple).